account_brand/models/account_invoice.py

```diff
@@ -1,9 +1,20 @@
 # Copyright (C) 2019 Open Source Integrators
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import models
+from odoo import api, models
 
 
 class AccountInvoice(models.Model):
     _name = 'account.invoice'
     _inherit = ['account.invoice', 'res.brand.mixin']
+
+    @api.onchange('brand_id', 'invoice_line_ids')
+    def _onchange_brand_id(self):
```

Comment:
Are you sure this is triggered in the case the invoice is created from a Sale Order, and the brand is carried over from there?

Reply:
This onchange must not be triggered at creation as the analytic account is set in the sale order an carried with the standard code.

We keep the possibility for the user to unset the analytic account in the sale order and have a branded invoice without any analytic.